virtcontainers/api.go

```diff
@@ -96,19 +96,19 @@ func createSandboxFromConfig(ctx context.Context, sandboxConfig SandboxConfig, f
 		}
 	}()
 
-	// Start the VM
-	if err = s.startVM(); err != nil {
-		return nil, err
-	}
-
 	// rollback to stop VM if error occurs
 	defer func() {
 		if err != nil {
 			s.stopVM()
 		}
 	}()
 
-	if err := s.getAndStoreGuestDetails(); err != nil {
+	// Start the VM
```

Comment:
I don't think it's a good idea to this move. Since the s.startVM() will do two things: launch the hypervisor process and start the sandbox in hypervisor, on the contrary, the s.stopVM() will do the stop sandbox in the hypervisor and then stop the hypervisor process. 

Suppose  it failed on launching the hypervisor process, then the defer function s.stopVM() will be redundant since there is no need to do the cleanup. On the other hand, if it failed on starting the sandbox in hypervisor, then the cleanup function of stop the sandbox in hypervisor will be failed since the sandbox failed to be started, and the s.stopVM() will return directly without trying to stop the hypervisor process. This isn't what we want.

Reply:
yes, it is what I said in the comment
```
Then the second should be fix , how about call stopSandbox() inside startVM do the cleanup
```